examples/fluids/src/grid_anisotropy_tensor.c

```diff
@@ -145,17 +142,18 @@ PetscErrorCode GridAnisotropyTensorCalculateCollocatedVector(Ceed ceed, User use
   // -- Get Pre-requisite things
   *num_comp_aniso = 7;
   PetscCall(DMGetDimension(user->dm, &dim));
-  CeedBasisGetNumQuadraturePoints1D(ceed_data->basis_q, &num_qpts_1d);
+  CeedBasisGetNumQuadraturePoints(ceed_data->basis_q, &num_qpts);
   CeedBasisGetNumNodes1D(ceed_data->basis_q, &num_nodes_1d);
   CeedElemRestrictionGetNumComponents(ceed_data->elem_restr_qd_i, &q_data_size);
+  PetscCall(GetRestrictionForDomain(ceed, user->dm, 0, 0, 0, 0, num_qpts, *num_comp_aniso, NULL, NULL, elem_restr_grid_aniso));
 
-  PetscCall(GetRestrictionForDomain(ceed, user->dm, 0, 0, 0, 0, num_qpts_1d, *num_comp_aniso, NULL, NULL, elem_restr_grid_aniso));
-
-  CeedInt Q_dim = CeedIntPow(num_qpts_1d, dim);
   CeedElemRestrictionGetNumElements(ceed_data->elem_restr_q, &loc_num_elem);
-  CeedElemRestrictionCreateStrided(ceed, loc_num_elem, Q_dim, *num_comp_aniso, *num_comp_aniso * loc_num_elem * Q_dim, CEED_STRIDES_BACKEND,
+  CeedElemRestrictionCreateStrided(ceed, loc_num_elem, num_qpts, *num_comp_aniso, *num_comp_aniso * loc_num_elem * num_qpts, CEED_STRIDES_BACKEND,
                                    elem_restr_grid_aniso);
 
+  // TODO Use CreateBasisFromPlex, but need set custom number of components
+  CeedInt num_qpts_1d;
+  CeedBasisGetNumQuadraturePoints1D(ceed_data->basis_q, &num_qpts_1d);
   CeedBasisCreateTensorH1Lagrange(ceed, dim, *num_comp_aniso, num_nodes_1d, num_qpts_1d, CEED_GAUSS, &basis_grid_aniso);
```

Comment:
Not sure how we'd get around this, but I don't want it to hold up this PR.

Reply:
If you want to use `CreateBasisFromPlex`, then the DMPlex FE should have the number of components you need. We create auxiliary DMs for extra stuff like this in Ratel

Reply:
Right, I create auxillary DMs for most other situations, except for this one, as there isn't a need for a intermediary DM otherwise. It seems odd to create an entire DM just to get the basis, but I don't see a problem with it either.

Reply:
...or I can actually read the code and realize A) we do have a DM built for this and B) that `basis_grad_aniso` isn't even used in this context, so I don't even need to build a basis.

Reply:
Agreed this should not hold up the PR.  Who marks this resolved?